### PR TITLE
Add one-line module documentation

### PR DESCRIPTION
**Pull request by meditans**

### CODE REVIEW

**Comment by 3noch** [comment]
Thanks!

File: backend/Rhyolite/Backend/App.hs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+-- | This module contains the implementation the view/viewselector machinery.
```
Comment (alexfmpe) [inline backend/Rhyolite/Backend/App.hs]:
```suggestion
-- | This module contains the implementation of the view/viewselector machinery.
```

File: datastructures/src/Rhyolite/SemiMap.hs [hunk 2]
```diff
@@ -21,9 +23,13 @@ import GHC.Generics (Generic)
 
 import Rhyolite.Aeson.Orphans ()
 
+-- | A SemiMap is a structure built on top on the `MonoidalMap` that let you
```
Comment (alexfmpe) [inline datastructures/src/Rhyolite/SemiMap.hs]:
```suggestion
-- | A SemiMap is a structure built on top on the `MonoidalMap` that lets you
```

File: frontend/Rhyolite/Frontend/Modal/Base.hs [hunk 3]
```diff
@@ -1,3 +1,6 @@
+-- | Modal widgets. The important definition here is `ModalT`, the relative
```
Comment (alexfmpe) [inline frontend/Rhyolite/Frontend/Modal/Base.hs]: Did you perhaps mean 'related' or 'respective'? I guess 'relative' technically makes sense but it sounds odd to me

Comment (3noch) [reply frontend/Rhyolite/Frontend/Modal/Base.hs]: Yes 'related' would be better.

**Comment by 3noch** [comment]
Please replace backticked ``` ` ``` references with single-quotes which is what Haddock uses.

File: backend/Rhyolite/Backend/App.hs (hunk 1 above)
Comment (meditans) [reply backend/Rhyolite/Backend/App.hs]: Thanks!

File: datastructures/src/Rhyolite/SemiMap.hs (hunk 2 above)
Comment (meditans) [reply datastructures/src/Rhyolite/SemiMap.hs]: Thanks!

File: frontend/Rhyolite/Frontend/Modal/Base.hs (hunk 3 above)
Comment (meditans) [reply frontend/Rhyolite/Frontend/Modal/Base.hs]: Thanks!

**Comment by meditans** [comment]
Corrections implemented, thanks!

**Comment by 3noch** [comment]
Thank you!

**Pull request merged**